power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -33,21 +33,30 @@ template <component_c Component, class ComponentContainer,
           std::output_iterator<Idx2D> OutputIterator>
     requires model_component_state_c<MainModelState, ComponentContainer, Component>
 inline void get_component_sequence(MainModelState<ComponentContainer> const& state, ForwardIterator begin,
-                                   ForwardIterator end, OutputIterator destination) {
+                                   ForwardIterator end, OutputIterator destination, Idx n_comp_elements) {
     using UpdateType = typename Component::UpdateType;
 
+    auto idx_getter_func = [&state, n_comp_elements](UpdateType const& update, auto index) {
+        if (n_comp_elements == na_Idx) {
```

Comment:
Why does `is_nan` live in `three_phase_sensor.hpp` ???

Reply:
`tensor`, not `sensor` 😉 the underlying reason is that it's better to have all `is_nan` functions in one location, rather than scattered around, and it needs to be defined for the 3p tensor.

then again: no real reason and if you have a better proposal that serves both the above reasons, then feel free to change it (but please do so in a separate PR to keep it clean)

Reply:
`sensor` to `tensor` doesn't change the merit of the question: such a fundamental function like `is_nan` should live somewhere else rather than where convenient. Will create an issue for that.

Reply:
This line in particular is not possible since "call to 'is_nan' is ambiguous". Will update once the new issue is fixed in another PR, i.e., `is_nan` moved to `common.hpp`.

Reply:
Interesting... in that case that's the way to go